k8s-operator/apis/v1alpha1/types_tsrecorder.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !plan9
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,shortName=rec
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=`.status.conditions[?(@.type == "RecorderReady")].reason`,description="Status of the deployed TSRecorder resources."
+
+type TSRecorder struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec describes the desired recorder instance.
+	Spec TSRecorderSpec `json:"spec"`
+
+	// TSRecorderStatus describes the status of the recorder. This is set
+	// and managed by the Tailscale operator.
+	// +optional
+	Status TSRecorderStatus `json:"status"`
+}
+
+// +kubebuilder:object:root=true
+
+type TSRecorderList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []TSRecorder `json:"items"`
+}
+
+type TSRecorderSpec struct {
+	// Tags that the Tailscale node will be tagged with. Defaults to [tag:k8s-recorder].
+	// If you specify custom tags here, make sure you also make the operator
+	// an owner of these tags.
+	// See  https://tailscale.com/kb/1236/kubernetes-operator/#setting-up-the-kubernetes-operator.
+	// Tags cannot be changed once a TSRecorder node has been created.
+	// Tag values must be in form ^tag:[a-zA-Z][a-zA-Z0-9-]*$.
+	// +optional
+	Tags Tags `json:"tags,omitempty"`
+
+	// TODO(tomhjp): Support a hostname or hostname prefix field, depending on
+	// the plan for multiple replicas.
+
+	// TSRecorder image. Defaults to tailscale/tsrecorder, with the same tag as
+	// the operator.
+	// +optional
+	Image Image `json:"image,omitempty"`
+
+	// If enabled, TSRecorder will serve the UI with HTTPS on its MagicDNS hostname.
+	// See --ui flag for more details: https://tailscale.com/kb/1246/tailscale-ssh-session-recording#deploy-a-recorder-node.
+	// +optional
+	EnableUI bool `json:"enableUI,omitempty"`
+
+	// Additional volumes for the pod spec. May be useful if you want to use a
+	// local file path for storage. For more details, see --dst flag:
+	// https://tailscale.com/kb/1246/tailscale-ssh-session-recording#deploy-a-recorder-node.
+	// +optional
+	ExtraVolumes []corev1.Volume `json:"extraVolumes,omitempty"`
```

Comment:
👍 While I was doing that, it seemed to make sense to also just copy some of the other fields of ProxyClass too, so that there's better consistency between the two resources.